### PR TITLE
Support mmap shared anonymous memory 

### PR DESCRIPTION
**Pull request by StevenJiang1110**

### CODE REVIEW

**Comment by tatetian** [comment]
@cchanging Could you give some comments on this PR? A lot have changed in VMO since this PR is submitted.

File: kernel/aster-nix/src/vm/vmar/vm_mapping.rs [hunk 1]
```diff
@@ -24,6 +24,8 @@ pub struct VmMapping {
     parent: Weak<Vmar_>,
     /// The mapped vmo. The mapped vmo is with dynamic capability.
     vmo: Vmo<Rights>,
+    /// The mapping is shared among processes
+    is_shared: bool,
```
Comment (cchanging) [inline kernel/aster-nix/src/vm/vmar/vm_mapping.rs:28]: -> Whether the mapping is shared among processes

The `is_shared` in `VmMapping` can also indicate that modifications to the mapping in the case of file-backed scenarios will be reflected in the underlying file. I think we can add a TODO to the doc here to reflect this situation and indicate we do not support it yet.

**Review by cchanging — COMMENTED** [review]
This PR looks good to me. It adds `is_shared` flag in `VmMapping` to support shared anounymous mmap which is almost decoupled to the vmo-related modifications before. 

I think this PR can be compatible with current vm modules. It just needs to rebase.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**